tests/integration/devfile/cmd_devfile_deploy_test.go

```diff
@@ -0,0 +1,181 @@
+package devfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo devfile deploy command tests", func() {
+	var namespace, context, cmpName, currentWorkingDirectory, originalKubeconfig, imageTag string
+	// Using program commmand according to cliRunner in devfile
+	cliRunner := helper.GetCliRunner()
+	oc := helper.NewOcRunner("oc")
+
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		context = helper.CreateNewContext()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+
+		// Devfile push requires experimental mode to be set
+		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+
+		originalKubeconfig = os.Getenv("KUBECONFIG")
+		helper.LocalKubeconfigSet(context)
+		namespace = cliRunner.CreateRandNamespaceProject()
+		imageTag = fmt.Sprintf("image-registry.openshift-image-registry.svc:5000/%s/my-nodejs:1.0", namespace)
+		currentWorkingDirectory = helper.Getwd()
+		cmpName = helper.RandString(6)
+		helper.Chdir(context)
+	})
+
+	var _ = AfterEach(func() {
+		cliRunner.DeleteNamespaceProject(namespace)
+		helper.Chdir(currentWorkingDirectory)
+		err := os.Setenv("KUBECONFIG", originalKubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("Verify deploy completes when passing a valid Dockerfile URL from the devfile", func() {
+		It("Should succesfully download the dockerfile and build the project", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfilesV2", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.CmdShouldPass("odo", "url", "create", "--port", "3000")
+			err := helper.ReplaceDevfileField("devfile.yaml", "alpha.deployment-manifest",
+				fmt.Sprintf("file://%s/../../examples/source/manifests/deploy_deployment_clusterip.yaml", currentWorkingDirectory))
+			Expect(err).To(BeNil())
+			output := helper.CmdShouldPass("odo", "deploy", "--tag", imageTag)
+			cliRunner.WaitAndCheckForExistence("buildconfig", namespace, 1)
+			Expect(output).NotTo(ContainSubstring("does not point to a valid Dockerfile"))
+			Expect(output).To(ContainSubstring("Successfully built container image"))
+			Expect(output).To(ContainSubstring("Successfully deployed component"))
```

Comment:
Merge these two lines into a single statement. For example  https://github.com/openshift/odo/pull/3478/files#diff-970a8865c1ac3e92e417588538f6b10fR60